### ISSUE TITLE
v1.10-dpr: PHP Fatal error:  Interface 'TemplateVariable' not found 

### ISSUE THREAD

**Issue by infectormp** [osTicket/osTicket-1.8#2060]
I stock on first step - installation
Get this message when try open new site for install

PHP Fatal error:  Interface 'TemplateVariable' not found in /xxx/include/class.format.php on line 808

**Comment by mfelber** [comment]
The same happens when I try to install v1.10-dpr: white page is showing and in the error log I see the same error.

**Comment by Aka4** [comment]
same here, white page is showing and in the error log I see:
PHP Fatal error:  Interface 'TemplateVariable' not found in .../include/class.format.php on line 808

**Comment by Chefkeks** [comment]
Just out of curiosity: From where did you guys download the 1.10 DPR ?
Github or osTicket.com/download

Note: Have not tried 1.10 DPR yet.

**Comment by infectormp** [comment]
@Chefkeks 1.10 DPR available only at github

**Comment by mfelber** [comment]
@Chefkeks Both: I have tried Github and osTicket.com/download same error occurs in both cases.

**Comment by mfelber** [comment]
@infectormp 1.10 DPR is already available at osticket.com/download:
http://osticket.com/download/go?dl=osTicket-v1.10-dpr.zip

**Comment by infectormp** [comment]
@mfelber good

**Comment by Chefkeks** [comment]
Ok I guess @greezybacon and @protich will take care of that.

Going to try it out tomorrow since today is a holiday here in Germany which I'll spend eating strawberry cake, etc. to be fully relaxed for DPR test on Friday then :D

**Comment by simonebracaloni** [comment]
Hi everione.
 To solve this issue I had to add a require  (require_once(INCLUDE_DIR.'class.variable.php'); ) in setup.inc.php.

#required files
require_once(INCLUDE_DIR.'class.setup.php');
require_once(INCLUDE_DIR.'class.validator.php');
require_once(INCLUDE_DIR.'class.passwd.php');
require_once(INCLUDE_DIR.'class.variable.php');
require_once(INCLUDE_DIR.'class.format.php');
require_once(INCLUDE_DIR.'class.misc.php');
require_once INCLUDE_DIR.'mysqli.php';
require_once INCLUDE_DIR.'class.i18n.php';

**Comment by mfelber** [comment]
I have seen this error first in the develop-next branch. My Testinstallation from second may doesn't have this error. I have tested some days later again (7th may) with the actual develop-next and this error occurs when I tried to install. So the change happened between second may and 7th may. Since this time I couldn't install the develop-next no more because of this error. This is the error message from the 7th may which is very similar: Fatal error: Interface 'TemplateVariable' not found in var/www/html/osticket/include/class.format.php on line 805. Maybe this helps to investigate this..

@Chefkeks This sounds to me a good plan ;-)

**Comment by mfelber** [comment]
@simonebracaloni Yeah, this change makes it possible to install. Thanks :-)

**Comment by infectormp** [comment]
hm, dosn't help for me

**Comment by mfelber** [comment]
@infectormp Maybe this is a browser cache issue. After the changes it doesn't worked for me too, but after I cleared all in my browser it works..

**Comment by infectormp** [comment]
not sure, i try differect browser and i still have same error in apache logs

**Comment by fgenellini** [comment]
@simonebracaloni  Thank you, that if it works!!

**Comment by greezybacon** [comment]
Thanks guys! I fixed this in 3e4b685

**Comment by greezybacon** [comment]
I've posted a new download on osTicket.com and on the Releases page on GitHub which corrects the issue.

**Comment by mfelber** [comment]
@greezybacon I have tested both downloads again and now the installation works fine :-)

**Comment by greezybacon** [comment]
:+1:

**Comment by harrisyn** [comment]
Odd, that I am unable to install this, I enter all the correct parameters, then it creates the database but not the tables, and gives me a white screen.

Any ideas where to look? or where the logs are kept?

**Comment by greezybacon** [comment]
@harrisyn that's specific to your server. Please don't attempt installing with a language pack. Install US-English and then configure the languages afterwards

**Comment by harrisyn** [comment]
@greezybacon all I need is english, its the only option I have or need.
previous versions have worked without issues, stomped on this one.

**Comment by mfelber** [comment]
@harrisyn I think the behavior you describe is a different issue. In this issue we weren't able to come to the installation page. According to your description your problem occurs later. Maybe you should start a different issue report because it seems to have a different cause. If I'm not wrong the white screen indicates an php error. So you should look in your php error logs of your web server. The location depends on which path you (or your hoster?) have chosen. The osticket logs you can find under "Admin panel"->"Dashboard"->"System Logs".

**Comment by bateller** [comment]
Just FYI I just downloaded the v1.10 developer preview from https://github.com/osTicket/osTicket-1.8/releases/tag/v1.10-dpr

The Source code (tar.gz) version from here (https://github.com/osTicket/osTicket-1.8/archive/v1.10-dpr.tar.gz) still had this issue. Adding the required statement for class.variable.php fixed the issue. I didn't see any note on the Release page detailing the error that @greezybacon mentioned was added. Its possible I'm just missing something or the error is isolated to the tar.gz version.
